pkg/controller/trainjob_controller.go

```diff
@@ -83,54 +75,70 @@ func (r *TrainJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
+	var err error
+	// Keep track of the origin TrainJob status
+	originStatus := trainJob.Status.DeepCopy()
+
+	// Let's clear the failed condition that could have been set previously.
+	// An external change to the TrainJob spec may transition it out of the Failed state.
+	removeFailedCondition(&trainJob)
+
 	runtimeRefGK := jobruntimes.RuntimeRefToRuntimeRegistryKey(trainJob.Spec.RuntimeRef)
 	runtime, ok := r.runtimes[runtimeRefGK]
 	if !ok {
-		return ctrl.Result{}, fmt.Errorf("unsupported runtime: %s", runtimeRefGK)
+		err = fmt.Errorf("unsupported runtime: %s", runtimeRefGK)
+		setFailedCondition(&trainJob, fmt.Sprintf("unsupported runtime: %s", runtimeRefGK), trainer.TrainJobRuntimeNotSupportedReason)
+	} else {
+		err = r.reconcileObjects(ctx, runtime, &trainJob)
+		if err != nil {
+			// TODO (astefanutti): the error should be surfaced in the TrainJob status to indicate
+			//  the creation of the runtime resources failed and the TrainJob is backed off until
+			//  the next retry attempt.
+			r.recorder.Eventf(&trainJob, corev1.EventTypeWarning, "TrainJobResourcesCreationFailed",
+				"TrainJob resources reconciliation failed: %.1024v", err.Error())
```

Comment:
```suggestion
			msg = fmt.Sprintf("TrainJob resources reconciliation failed: %.950v", err.Error())
			if len(err.Error()) > 1024 {
				msg = fmt.Sprintf("%s...", msg)
			}
			r.recorder.Eventf(&trainJob, corev1.EventTypeWarning, "TrainJobResourcesCreationFailed", msg)
```

The entire message size must be 1024 and `TrainJob resources reconciliation failed` is 42 words. So, Let's truncate it as 950.

Reply:
That's a great point, thanks! Updated according to your suggestion. PTAL.